src/npcmove.cpp

```diff
@@ -3016,13 +3017,27 @@ void npc::set_destination()
     tripoint surface_omt_loc = global_omt_location();
     surface_omt_loc.z = 0;
 
-    std::string dest_type = get_location_for( needs.front() )->get_random_terrain().id().str();
-    goal = overmap_buffer.find_closest( surface_omt_loc, dest_type, 0, false );
+    std::string dest_type, current_dest_type;
+    tripoint current_goal;
+    int current_goal_dist, goal_dist = -1;
```

Comment:
Style issue, each variable goes on its own line.
```suggestion
    int current_goal_dist = 0;
    int goal_dist = -1;
```